recipes-qt/qt5/qt5-qpa-hwcomposer-plugin/0001-Synchronize-time-on-LCD.patch

```diff
@@ -0,0 +1,144 @@
+From 0bb6ac637b793ff8c44a4eef210213c033ff24ab Mon Sep 17 00:00:00 2001
+From: MagneFire <dgriet@gmail.com>
+Date: Sat, 22 Jan 2022 16:13:26 +0100
+Subject: [PATCH] Synchronize time on LCD.
+Use the vendor provided libmcutool.so library to synchronize the LCD time when the screen turns off.
+
+---
+ hwcomposer/hwcomposer_backend_v11.cpp | 57 +++++++++++++++++++++++++++
+ hwcomposer/hwcomposer_backend_v11.h   | 17 ++++++++
+ 2 files changed, 74 insertions(+)
+
+diff --git a/hwcomposer/hwcomposer_backend_v11.cpp b/hwcomposer/hwcomposer_backend_v11.cpp
+index 8f158cd..76b3a49 100644
+--- a/hwcomposer/hwcomposer_backend_v11.cpp
++++ b/hwcomposer/hwcomposer_backend_v11.cpp
+@@ -48,12 +48,47 @@
+ #include <QtCore/QCoreApplication>
+ #include <private/qwindow_p.h>
+ 
++//#include <android-config.h>
++//#include <hardware/hardware.h>
++//#include <hybris/common/binding.h>
++#include <dlfcn.h>
++
+ #include "qsystrace_selector.h"
+ 
+ #ifdef HWC_PLUGIN_HAVE_HWCOMPOSER1_API
+ 
+ // #define QPA_HWC_TIMING
+ 
++
++#define LOAD_MCU_FUNC(x) { \
++    mcu->x = (int32_t (*)())android_dlsym(handler, "Java_com_mobvoi_ticwear_mcuservice_CoreService_"#x); \
++    if (mcu->x == NULL) { \
++        fprintf(stderr, "Unable to get symbol\n"); \
++    } \
++}
++#define LOAD_MCU_FUNC1(x) { \
++    mcu->x = (int32_t (*)(int32_t))android_dlsym(handler, "Java_com_mobvoi_ticwear_mcuservice_CoreService_"#x); \
++    if (mcu->x == NULL) { \
++        fprintf(stderr, "Unable to get symbol\n"); \
++    } \
++}
++#define LOAD_MCU_FUNC3(x) { \
++    mcu->x = (int32_t (*)(int32_t, int32_t, int32_t))android_dlsym(handler, "Java_com_mobvoi_ticwear_mcuservice_CoreService_"#x); \
++    if (mcu->x == NULL) { \
++        fprintf(stderr, "Unable to get symbol\n"); \
++    } \
++}
++#define LOAD_MCU_FUNC5(x) { \
```

Comment:
Ooc have you tried using variadic macros ? something along the line of https://github.com/torvalds/linux/blob/master/tools/lib/bpf/bpf_tracing.h#L321